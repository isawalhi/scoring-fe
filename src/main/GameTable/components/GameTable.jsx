import React from 'react';
import { Link } from 'react-router-dom';
import { Table as AntTable } from 'antd';

const GameTable = ({ sortedPlayers, loading }) => {
  const columns = [
    {
      title: 'Player Name',
      dataIndex: 'name',
      key: 'name',
      // TODO get path from routes
      // eslint-disable-next-line react/display-name
      render: (text) => <Link to={`player/${text}`}>{text}</Link>,
    },
    {
      title: 'Number of Games',
      dataIndex: 'numberOfGames',
      key: 'numberOfGames',
    },
    {
      title: 'Score',
      dataIndex: 'score',
      key: 'score',
    },
  ];
  return (
    <AntTable
      pagination={{ pageSize: 10 }}
      columns={columns}
      dataSource={sortedPlayers}
      loading={sortedPlayers.length === 0 && loading}
    />
  );
};

export default GameTable;
