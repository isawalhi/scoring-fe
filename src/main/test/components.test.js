import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GameTable from '../GameTable/components/GameTable';

describe('<GameTable />', () => {
  const players = [
    {
      name: 'ibrahim',
      score: 1400,
      scoreHistory: [{}],
      namberOfGames: 1,
    },
    {
      name: 'batoul',
      score: 1600,
      scoreHistory: [{}],
      namberOfGames: 1,
    },
  ];
  const props = {
    loading: true,
    sortedPlayers: [],
  };

  const wrapper = shallow(<GameTable {...props} />);

  it('renders correctly - loading true - empty table', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly - loading false - empty table', () => {
    wrapper.setProps({ loading: false });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly - loading false - table with data ', () => {
    wrapper.setProps({ sortedPlayers: players });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly - loading true - table with data ', () => {
    wrapper.setProps({ loading: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
