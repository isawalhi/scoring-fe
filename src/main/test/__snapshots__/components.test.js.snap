// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameTable /> renders correctly - loading false - empty table 1`] = `
<Table
  columns={
    Array [
      Object {
        "dataIndex": "name",
        "key": "name",
        "render": [Function],
        "title": "Player Name",
      },
      Object {
        "dataIndex": "numberOfGames",
        "key": "numberOfGames",
        "title": "Number of Games",
      },
      Object {
        "dataIndex": "score",
        "key": "score",
        "title": "Score",
      },
    ]
  }
  dataSource={Array []}
  loading={false}
  pagination={
    Object {
      "pageSize": 10,
    }
  }
  rowKey="key"
/>
`;

exports[`<GameTable /> renders correctly - loading false - table with data  1`] = `
<Table
  columns={
    Array [
      Object {
        "dataIndex": "name",
        "key": "name",
        "render": [Function],
        "title": "Player Name",
      },
      Object {
        "dataIndex": "numberOfGames",
        "key": "numberOfGames",
        "title": "Number of Games",
      },
      Object {
        "dataIndex": "score",
        "key": "score",
        "title": "Score",
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "namberOfGames": 1,
        "name": "ibrahim",
        "score": 1400,
        "scoreHistory": Array [
          Object {},
        ],
      },
      Object {
        "namberOfGames": 1,
        "name": "batoul",
        "score": 1600,
        "scoreHistory": Array [
          Object {},
        ],
      },
    ]
  }
  loading={false}
  pagination={
    Object {
      "pageSize": 10,
    }
  }
  rowKey="key"
/>
`;

exports[`<GameTable /> renders correctly - loading true - empty table 1`] = `
<Table
  columns={
    Array [
      Object {
        "dataIndex": "name",
        "key": "name",
        "render": [Function],
        "title": "Player Name",
      },
      Object {
        "dataIndex": "numberOfGames",
        "key": "numberOfGames",
        "title": "Number of Games",
      },
      Object {
        "dataIndex": "score",
        "key": "score",
        "title": "Score",
      },
    ]
  }
  dataSource={Array []}
  loading={true}
  pagination={
    Object {
      "pageSize": 10,
    }
  }
  rowKey="key"
/>
`;

exports[`<GameTable /> renders correctly - loading true - table with data  1`] = `
<Table
  columns={
    Array [
      Object {
        "dataIndex": "name",
        "key": "name",
        "render": [Function],
        "title": "Player Name",
      },
      Object {
        "dataIndex": "numberOfGames",
        "key": "numberOfGames",
        "title": "Number of Games",
      },
      Object {
        "dataIndex": "score",
        "key": "score",
        "title": "Score",
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "namberOfGames": 1,
        "name": "ibrahim",
        "score": 1400,
        "scoreHistory": Array [
          Object {},
        ],
      },
      Object {
        "namberOfGames": 1,
        "name": "batoul",
        "score": 1600,
        "scoreHistory": Array [
          Object {},
        ],
      },
    ]
  }
  loading={false}
  pagination={
    Object {
      "pageSize": 10,
    }
  }
  rowKey="key"
/>
`;
